packages/frontend/src/helpers/variables.ts

```diff
@@ -35,28 +35,35 @@ function postProcess(
   metadata: IDataOutMetadata,
 ): Variable[] {
   const result: Variable[] = []
-
   for (const variable of variables) {
     const { name, ...rest } = variable
+
     // lodash get does its work by specifying the 'name' path
     const {
       isHidden = false,
       type = null,
-      label = null,
       order = null,
       displayedValue = null,
     } = get(metadata, name, {}) as IDataOutMetadatum
 
-    // the fields with .order will be hidden and have no output content (filtered later)
+    // only label may change if metadata exists
+    let { label = null } = get(metadata, name, {}) as IDataOutMetadatum
+
+    // the fields with .order and .fieldType will be hidden and have no output content (filtered later)
```

Comment:
We should remove these comments; the fields you mentioned only apply to FormSG, but this function is called for every action and trigger.

Anyway it's pretty self-explanatory.